corehq/ex-submodules/phonelog/utils.py

```diff
@@ -42,11 +43,13 @@ def _get_logs(form, report_name, report_slug):
 
 
 @transaction.atomic
-def process_device_log(domain, xform):
+def process_device_log(domain, xform, force_logs):
     _process_user_subreport(xform)
     _process_log_subreport(domain, xform)
     _process_user_error_subreport(domain, xform)
     _process_force_close_subreport(domain, xform)
+    if force_logs:
+        clear_device_log_request(domain, xform)
```

Comment:
I also considered moving this call further up the stack, would be interested in second opinions on where this belongs.